currency_monthly_rate/models/res_currency.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2018 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models, fields, api, _
+
+
+class ResCurrency(models.Model):
+    _inherit = "res.currency"
+
+    monthly_rate = fields.Float(compute='_compute_current_monthly_rate',
+                                string='Current Monthly Rate', digits=(12, 6),
+                                help='The monthly rate of the currency to '
+                                     'the currency of rate 1.')
+    monthly_rate_ids = fields.One2many('res.currency.rate.monthly',
+                                       'currency_id', string='Monthly rates')
+
+    @api.multi
+    def _compute_current_monthly_rate(self):
+        date = self.env.context.get('date') or fields.Date.today()
+        company_id = self.env.context.get('company_id') or self.env[
+            'res.users']._get_company().id
```

Comment:
The standard uses the wrapper method `self.env['res.company']._company_default_get()`

Reply:
Actually not, so I suppose it make sense to check the company in context first :
https://github.com/odoo/odoo/blob/11.0/odoo/addons/base/res/res_currency.py#L49